src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/DebugInfoTableNode.cs

```diff
@@ -206,6 +209,19 @@ public static byte[] CreateVarBlobForMethod(NativeVarInfo[] varInfos)
                     default:
                         throw new BadImageFormatException("Unexpected var loc type");
                 }
+
+                static void WriteEncodedStackOffset(NibbleWriter _writer, int offset, bool assume4ByteAligned)
```

Comment:
Nit - if at some point we considered expanding this behavior to other architectures, the name of the argument (<code>assume4ByteAligned</code>) would probably need to change to e.g. something like <code>encodeUsingPointerSizeUnits</code> or some such. If we believe this to be limited to X86 forever, it's probably OK.

Reply:
I just checked to see how valuable this really is, and we could save something like 0.08% of System.Private.CoreLib.dll size if we applied a similar size optimization on X64. I don't see that as worthwhile given the complexity it would impose on the reader in the runtime.